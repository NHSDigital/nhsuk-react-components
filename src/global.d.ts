declare module 'nhsuk-frontend/packages/components/header/header.js';
declare module 'nhsuk-frontend/packages/components/checkboxes/checkboxes.js';
declare module 'nhsuk-frontend/packages/components/character-count/character-count.js';
declare module 'nhsuk-frontend/packages/components/tabs/tabs.js';
