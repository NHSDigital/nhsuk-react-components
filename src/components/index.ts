export * from './content-presentation/index.js';
export * from './form-elements/index.js';
export * from './layout/index.js';
export * from './navigation/index.js';
export * from './typography/index.js';
export * from './utils/index.js';
