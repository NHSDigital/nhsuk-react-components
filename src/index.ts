export * from './components';
export * from './patterns';
