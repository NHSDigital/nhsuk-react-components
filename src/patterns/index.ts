export * from './nav-a-z/index.js';
export * from './review-date/index.js';
