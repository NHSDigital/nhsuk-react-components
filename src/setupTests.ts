import { TextDecoder, TextEncoder } from 'util';
import '@testing-library/jest-dom';
import { outdent } from 'outdent';

/**
 * Polyfill TextEncoder/TextDecoder for ReactDOM
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(globalThis, {
  TextEncoder: {
    value: TextEncoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
  TextDecoder: {
    value: TextDecoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
});

/**
 * Polyfill Element methods for NHS.UK frontend error summary
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(Element.prototype, {
  scrollIntoView: { value: jest.fn() },
});

/**
 * Polyfill `matchMedia()` for NHS.UK frontend tabs
 */
Object.defineProperty(globalThis, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    addEventListener: jest.fn(),
    removeListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

beforeEach(() => {
  const stylesheet = document.createElement('style');

  stylesheet.innerHTML = outdent`
    :root {
      --nhsuk-breakpoint-mobile: 20rem;
      --nhsuk-breakpoint-tablet: 40.0625rem;
      --nhsuk-breakpoint-desktop: 48.0625rem;
      --nhsuk-breakpoint-large-desktop: 61.875rem;
    }
  `;

  // Add styles for NHS.UK frontend checks
  document.head.appendChild(stylesheet);

  // Flag NHS.UK frontend as supported
  document.body.classList.add('nhsuk-frontend-supported');
});
