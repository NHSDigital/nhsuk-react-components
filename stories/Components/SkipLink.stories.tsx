import React from 'react';
import { SkipLink, Hint } from '../../src';
import { Meta, StoryObj } from '@storybook/react';

const CodeText: React.FC = (props) => (
  <span
    style={{
      fontFamily: 'monospace',
      marginLeft: 5,
      marginRight: 5,
      fontSize: 14,
      padding: 3,
      backgroundColor: 'rgba(255, 0, 0, 0.15)',
      borderRadius: 5,
    }}
    {...props}
  />
);

const meta: Meta<typeof SkipLink> = {
  title: 'Components/SkipLink',
  component: SkipLink,
  argTypes: {
    focusTargetRef: { table: { disable: true } },
  },
  render: (args) => (
    <>
      <Hint>
        Press
        <CodeText>tab</CodeText>
        to show the SkipLink
      </Hint>
      <SkipLink disableDefaultBehaviour={args.disableDefaultBehaviour} />
    </>
  ),
};
export default meta;
type Story = StoryObj<typeof SkipLink>;

export const Standard: Story = {
  args: {
    disableDefaultBehaviour: false,
  },
  argTypes: {
    disableDefaultBehaviour: { table: { disable: true } },
  },
};

export const SkipLinkWithDefaultBehaviourDisabled: Story = {
  args: {
    disableDefaultBehaviour: true,
  },
};
