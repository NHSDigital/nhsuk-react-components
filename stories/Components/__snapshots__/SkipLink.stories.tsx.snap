// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/SkipLink SkipLinkWithDefaultBehaviourDisabled smoke-test 1`] = `
<div class="nhsuk-hint">
  Press
  <span style="font-family: monospace; margin-left: 5px; margin-right: 5px; font-size: 14px; padding: 3px; background-color: rgba(255, 0, 0, 0.15); border-radius: 5px;">
    tab
  </span>
  to show the SkipLink
</div>
<a class="nhsuk-skip-link"
   tabindex="0"
>
  Skip to main content
</a>
`;

exports[`Components/SkipLink Standard smoke-test 1`] = `
<div class="nhsuk-hint">
  Press
  <span style="font-family: monospace; margin-left: 5px; margin-right: 5px; font-size: 14px; padding: 3px; background-color: rgba(255, 0, 0, 0.15); border-radius: 5px;">
    tab
  </span>
  to show the SkipLink
</div>
<a class="nhsuk-skip-link"
   href="#maincontent"
   tabindex="0"
>
  Skip to main content
</a>
`;
